PersonStageThree/src/com/beyond/person/core/model/BasePerson.java

```diff
@@ -0,0 +1,134 @@
+package com.beyond.person.core.model;
+
+import com.beyond.person.core.exception.InvalidAgeException;
+import com.beyond.person.core.exception.InvalidEmailException;
+import com.beyond.person.core.exception.InvalidFieldException;
+import com.beyond.person.core.exception.InvalidGenderException;
+import com.beyond.person.core.utils.Gender;
+
+public class BasePerson {
+    protected String name;
+    protected String lastName;
+    protected String nickname;
+    protected String email;
+    protected String designation;
+    protected String gender;
+    protected int age;
+    protected String email_regex = "^(.+)@(.+)$";
+
+    /**
+     * constructor for base person
+     * @param name
+     * @param lastName
+     */
+    public BasePerson(String name, String lastName) {
+        try {
+            if (0 < name.length() && name.length() < 255 && 0 < lastName.length() && lastName.length() < 255) {
+                this.name = name;
+                this.lastName = lastName;
+            } else {
+                throw new InvalidFieldException();
+            }
+        } catch (InvalidFieldException e) {
```

Comment:
@maria-salmasian why you catch your throw exception ?

Reply:
changed it